src/CppAst/CppTemplateArgument.cs

```diff
@@ -77,6 +89,37 @@ public override int SizeOf
             set => throw new InvalidOperationException("This type does not support SizeOf");
         }
 
+        public bool Equals(CppTemplateArgument other)
+        {
+            var result = base.Equals(other) &&

```

Comment:
This can go after all the other checks and simply `return base.Equals(other) && ...`